controllers/managedfusiondeployment_controller.go

```diff
@@ -0,0 +1,446 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	opv1a1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	controller "sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/go-logr/logr"
+	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	promv1a1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+	"github.com/red-hat-storage/managed-fusion-agent/templates"
+	"github.com/red-hat-storage/managed-fusion-agent/utils"
+)
+
+const (
+	managedFusionAgentFinalizer          = "managedfusionagent.ibm.com/finalizer"
+	managedFusionAgentSecretName         = "managed-fusion-agent-config"
+	managedFusionAgentSecretSMTPKey      = "smtp_config"
+	managedFusionAgentSecretPagerDutyKey = "pager_duty_config"
+	prometheusName                       = "managed-fusion-prometheus"
+	alertmanagerName                     = "managed-fusion-alertmanager"
+	alertmanagerConfigName               = "managed-fusion-alertmanager-config"
+	alertmanagerConfigSecretName         = "managed-fusion-alertmanager-config"
+	agentCSVPrefix                       = "managed-fusion-agent"
+	monLabelKey                          = "app"
+	monLabelValue                        = "managed-fusion-agent"
+	pagerDutyKey                         = "pagerDutyServiceKey"
+	smtpPasswordKey                      = "smtpAuthPassword"
```

Comment:
If these are meant to hold the names of the secret data keys inside the alertmanagerSecret then the correct name should be `alertmanagerSecretPagerDutyServiceKeyKey` and `alertmanagerSecretSMTPAuthPasswordKey. But as these are very long names we could shorten them to `amSecretPagerDutyServiceKeyKey` and `amSecretSMTPAuthPasswordKey`